composer.json

```diff
@@ -18,19 +20,30 @@
     "minimum-stability": "stable",
 
     "require": {
-        "php": ">=5.5.0",
+        "php": ">=7.1",
         "ext-curl": "*",
         "ext-json": "*",
         "ext-mbstring": "*",
         "guzzlehttp/guzzle": "^6.3"
     },
     "require-dev": {
-        "phpunit/phpunit": "^4.8",
-        "phpdocumentor/phpdocumentor": "^2.3.1"
+        "phpunit/phpunit": "^7",
+        "squizlabs/php_codesniffer": "^3.2"
     },
     "extra": {
         "branch-alias": {
             "dev-master": "1.0.x-dev"
         }
+    },
+    "scripts": {
+        "check": [
```

Comment:
When do you trigger check scripts? Maybe need to be triggered on 'post-install-cmd' command

Reply:
The scripts are convenient for running tests and checks on the command line and from Travis for example, i.e. `$ composer check` runs CS checker then unit tests.

Reply:
is it automatically started? might be convenient to run it before each publish on composer also I guess.

Reply:
It's not automatic. There are different event names for automatic scripts but these are only executed for the root package, not dependencies (I think) https://getcomposer.org/doc/articles/scripts.md

Reply:
so we can't trigger it during 'post-install-cmd' or this kind of event. maybe 'post-install-cmd' is not the right one but you get the idea. Just for deployment process